File: go/rain_drops/rain_drops.go
package raindrops

import "fmt"

func Convert(number int) string {
	sound := ""

	if number%3 == 0 {
		sound += "Pling"
	}
	if number%5 == 0 {
		sound += "Plang"
	}
	if number%7 == 0 {
		sound += "Plong"
	}

	if sound == "" {
		// If the sound is empty then the number doesnt have 3, 5 or 7 as factors
		return fmt.Sprintf("%d", number)
	}

	return sound
}

File: rust/raindrops.rs
pub fn raindrops(n: u32) -> String {
    let drops: Vec<(u32, &str)> = vec![
        (3, "Pling"),
        (5, "Plang"),
        (7, "Plong"),
    ];
    let mut raindrops: String = String::from("");

    for (drop: u32, sound: &str) in drops {
        if n % drop == 0 {
            raindrops.push_str(sound)
        }
    }

    if raindrops == "" {
        return n.to_string();
    }

    return raindrops;
}
